wremnants/muon_selections.py

```diff
@@ -51,6 +61,26 @@ def define_trigger_muons(df):
 
     return df
 
+def define_muon_uT_variable(df, isWorZ, smooth3dsf=False, colNamePrefix="goodMuons"):    
+    if smooth3dsf:
+        if isWorZ:
+            df = theory_tools.define_prefsr_vars(df)
+            df = df.Define(f"{colNamePrefix}_uT0", f"wrem::zqtproj0_boson({colNamePrefix}_pt0, {colNamePrefix}_phi0, ptVgen, phiVgen)")
+        else:
+            # for background processes (Top and Diboson, since Wtaunu and Ztautau are part of isW or isZ)
+            # sum all gen e, mu, tau, or neutrinos to define the boson proxy
+            # choose particles with status 1 (stable) and statusFlag & 1 (prompt) or taus with status 2 (decayed)
+            # there is no double counting for leptons from tau decays, since they have status 1 but not statusFlag & 1
+            if "GenPart_leptonAndPhoton" not in df.GetColumnNames():
```

Comment:
Why not just use the 2D SFs for the not single V samples?

Reply:
Because you need multiple helpers and you also lose the correlation of uncertainties with signal, since at that point they will be independent. In addition, the 2D SF with integrate uT are also an approximation since the integration uses the W uT spectrum (for mT > 40 GeV), so for backgrounds you would already have a bias. In addition you get an even larger bias when you move to mT < 40 GeV (also for signal actually)

Reply:
ok, fair. Any idea how time consuming all the additional summing is? I guess probably not a big deal anyway